codegen/src/main/java/software/amazon/awssdk/codegen/poet/builder/BaseClientBuilderClass.java

```diff
@@ -177,6 +180,16 @@ private MethodSpec finalizeServiceConfigurationMethod() {
                          .addCode("interceptors = $T.mergeLists(interceptors, config.option($T.EXECUTION_INTERCEPTORS));\n",
                                   CollectionUtils.class, SdkClientOption.class);
 
+        if (model.getMetadata().isQueryProtocol()) {
+            builder.addStatement("$T<$T> protocolInterceptors = $T.singletonList(new $T())",
+                                 List.class,
+                                 ExecutionInterceptor.class,
+                                 Collections.class,
+                                 QueryParametersToBodyInterceptor.class);
+            builder.addStatement("interceptors = $T.mergeLists(interceptors, protocolInterceptors)",
```

Comment:
Does the order matter here? Intuitively I'd have thought the protocolInterceptors should come first.

Reply:
Yes, I do think the order matters and I agree that the argument for adding it first is valid. The reason to keep it last is that since the interceptor is somewhat invisible to the user,  if they add an interceptor of their own the results would be more consistent. I think adding it first might be more "right" but adding it last is more user-friendly.